apps/src/studio/samBatLogic.js

```diff
@@ -0,0 +1,137 @@
+var studioConstants = require('./constants');
+var Direction = studioConstants.Direction;
+var Position = studioConstants.Position;
+var codegen = require('../codegen');
+var api = require('./api');
+
+/**
+ * Interface for a set of custom game logic for playlab
+ * @param {Studio} studio Reference to global studio object
+ * @interface CustomGameLogic
+ */
+function CustomGameLogic(studio) {}
+
+/**
+ * Logic to be run once per playlab tick
+ *
+ * @function
+ * @name CustomGameLogic#onTick
+ */
+
+/**
+ * Custom logic for the Sam the Bat levels
+ * @constructor
+ * @implements CustomGameLogic
+ */
+var SamBatLogic = function (studio) {
+  this.studio_ = studio;
+  this.cached_ = {};
+  this.samIndex = 0;
+};
+
+SamBatLogic.prototype.onTick = function () {
+  this.sam = this.studio_.sprite[this.samIndex];
```

Comment:
In general, I like to predeclare any members I'm going to end up using in the constructor, so I'd do something like "this.sam = null" there, to make it easy to see that this is a member.